mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -463,12 +465,17 @@ class Word2VecModel private[spark] (
 
   // wordVecNorms: Array of length numWords, each value being the Euclidean norm
   //               of the wordVector.
-  private val wordVecNorms: Array[Double] = {
-    val wordVecNorms = new Array[Double](numWords)
+  private val wordVecNorms: Array[Float] = {
+    val wordVecNorms = new Array[Float](numWords)
     var i = 0
     while (i < numWords) {
       val vec = wordVectors.slice(i * vectorSize, i * vectorSize + vectorSize)
       wordVecNorms(i) = blas.snrm2(vectorSize, vec, 1)
+      // Normalize wordVectors while keeping the norm value in wordVecNorms
+      // in case the original value of wordVectors is needed
+      if(wordVecNorms(i) != 0) {
+        blas.sscal(vectorSize, 1 / wordVecNorms(i), wordVectors, i * vectorSize, 1)
```

Comment:
This changes the semantics of `getVectors`, since they will now be returned as normalized vectors. If we keep a normalized version internally in the model, then `getVectors` needs to be updated to return raw vectors.